deploy/index.js

```diff
@@ -0,0 +1,336 @@
+import { Octokit } from "octokit";
+import { writeFileSync } from "fs";
+import { tryPut, tryPutRemote } from "./fileaws.js";
+
+const MIME_JSON = "application/json";
+const MANIFEST_FILE_NAME = "manifest.json";
+const LATEST_FILE_NAME = "latest.json";
+
+const DEFAULT_JSON_INDENT = 2;
+const MIME_JAR = "application/java-archive";
+const MIME_TXT = "text/plain";
+
+const S3_BASE_URL = process.env.S3_BASE_URL;
+if (S3_BASE_URL === undefined) {
+  console.error("Error reading environment variable S3_BASE_URL.");
+  process.exit(4);
+}
+
+const TARGET_REPO = "/repos/" + process.env.GITHUB_REPOSITORY;
+if (TARGET_REPO === "/repos/") {
+  console.error("Error reading environment variable GITHUB_REPOSITORY.");
+  process.exit(5);
+}
+
+const REPO_OWNER = process.env.GITHUB_REPOSITORY_OWNER;
+if (REPO_OWNER === undefined) {
+  console.error("Error reading environment variable GITHUB_REPOSITORY_OWNER.");
+  process.exit(6);
+}
+
+const GITHUB_HEADERS = {
+  "X-GitHub-Api-Version": "2022-11-28",
+};
+
+/* Some tags should not be considered, if the tag_name includes any of the following strings, it should be filtered */
+const FILTER_TAG_STRINGS = ["pre", "beta", "alpha"];
+
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN,
+});
+
+/**
+ * Make Github API request for information on releases
+ *
+ * @param  owner - owner of the repo
+ * @param  repoName - name of the repo
+ * @returns Promise<OctokitResponse<any>>
+ */
+const makeGithubApi = async () => {
+  const resp = await octokit.request("GET " + TARGET_REPO + "/releases", {
+    owner: "OWNER",
+    repo: "REPO",
+    headers: GITHUB_HEADERS,
+  });
+  return resp;
+};
+
+/**
+ * Make Github API request for information on latest releases
+ *
+ * @param  owner - owner of the repo
+ * @param  repoName - name of the repo
```

Comment:
it looks like makeGithubApiLatestRequest does not take any parameters

Reply:
changed